File: 1437.check-if-all-1-s-are-at-least-length-k-places-away.py
#
# @lc app=leetcode id=1437 lang=python3
#
# [1437] Check If All 1's Are at Least Length K Places Away
#

# @lc code=start
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        if k == 0:
            return True
        distance = 0
        for n in nums:
            if n == 0:
                distance += 1
            else:
                if distance < 0:
                    return False
                else:
                    distance = -k
        return True
# @lc code=end

File: 1437.check-if-all-1-s-are-at-least-length-k-places-away.cpp
/*
 * @lc app=leetcode id=1437 lang=cpp
 *
 * [1437] Check If All 1's Are at Least Length K Places Away
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    bool kLengthApart(vector<int> &nums, int k)
    {
        if (k == 0)
        {
            return true;
        }
        auto pos = find(nums.begin(), nums.end(), 1);
        while (true)
        {
            auto next_pos = find(pos + 1, nums.end(), 1);
            if (next_pos == nums.end())
            {
                break;
            }
            else if (next_pos - pos <= k)
            {
                return false;
            }
            pos = next_pos;
        }
        return true;
    }
};
// @lc code=end
